experiments/experiment.go

```diff
@@ -380,6 +385,9 @@ func (ec *experimentContext) calculateStatus() *v1alpha1.ExperimentStatus {
 				// We will now fail the experiment.
 				ec.newStatus.Phase = analysesStatus
 				ec.newStatus.Message = analysesMessage
+			} else if analysesMessage == requiredAnalysisCompletedMessage {
```

Comment:
I think comparing the message against a well known message seems too magical. Is the point of this code to bubble up a better reason for a running experiment?

Reply:
That's a fair point. No, the purpose of this code is to provide a way to mark an experiment successful if the required analysis runs are completed. Since we shouldn't mark the experiment successful if the analysis is successful but not required, the magic string makes it easy to determine that case. Can you take another look?